src/lib.rs

```diff
@@ -4,24 +4,26 @@ pub fn hex_to_base64<T: ?Sized + AsRef<[u8]>>(input: &T) -> String {
 
     let mut output = String::new();
 
-    let byte_chunks = input.as_ref().chunks(3);
+    let mut bytes = input.as_ref().to_owned();
+    let byte_chunks = bytes.chunks_mut(3);
 
-    let mut empty_chunk_buffer = vec!(0u8,0u8,0u8);
+    let mut empty_chunk_buffer = vec![0, 0, 0];
 
     for chunk in byte_chunks {
-        let mut chunk_buffer = chunk.clone();
-        if chunk.len() > 3 {
-            chunk_buffer = chunk.iter().collect::<_>();
-            for (i, each) in (*chunk).iter().enumerate() {
-                chunk_buffer[i] = each.clone();
+        let chunk_buffer = if chunk.len() < 3 {
+            for (i, each) in chunk.iter().enumerate() {
+                empty_chunk_buffer[i] = *each;
             }
-        }
+            &empty_chunk_buffer[..]
+        } else {
+            &chunk[..]
+        };
```

Comment:
chunk_buffer is an immutable slice looking at the three bytes we are interested in.  for empty chunk buffer the &_[..] annotation creates a slice into the owned copy, for chunk we already have a mutable slice.  I'm not entirely sure, but I think rustc is smart enough to remove the mutability to match the type of `&empty_chunk_buffer[..]`  

The iter().collect::<_>(); piece on Red L14 above wasn't your code.  Someone added that at Triangle rustaceans, but it wasn't needed.

Reply:
I think this also works if you replace `&chunk[..]` with `chunk`